sdk/src/test/java/com/karhoo/sdk/api/service/quotes/CheckCoverageInteractorTest.kt

```diff
@@ -0,0 +1,116 @@
+package com.karhoo.sdk.api.service.quotes
+
+import com.karhoo.sdk.api.KarhooError
+import com.karhoo.sdk.api.model.Coverage
+import com.karhoo.sdk.api.network.request.CoverageRequest
+import com.karhoo.sdk.api.network.response.Resource
+import com.karhoo.sdk.api.testrunner.base.BaseKarhooUserInteractorTest
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+
+class CheckCoverageInteractorTest : BaseKarhooUserInteractorTest() {
+
+    internal lateinit var interactor: CheckCoverageInteractor
+
+    @Before
+    override fun setUp() {
+        super.setUp()
+        whenever(credentialsManager.isValidToken).thenReturn(true)
+        interactor = CheckCoverageInteractor(credentialsManager, apiTemplate, context)
+    }
+
+    /**
+     * Given:   A request is made to check fleet coverage of an area
+     * When:    There is coverage in the area
+     * Then:    A successful response is returned
+     **/
+    @Test
+    fun checkCoverageReturnsASuccessfulResponse() {
+        val latitude = "51.532156"
```

Comment:
You can probably set these up as const or even just create a CoverageRequest that can be reused.

Reply:
Done ✅